clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -1122,7 +1122,7 @@ public Generation(int generationId, String memberId, String protocol) {
         }
 
         public boolean isValid() {
-            return generationId != OffsetCommitRequest.DEFAULT_GENERATION_ID;
+            return this != NO_GENERATION;
```

Comment:
I think it is a bit confusing to call the generation valid if we are in the pending state. I am debating whether it is worth the complexity to add another `MemberState`, say `PENDING` for the state in which we have a memberId, but are not yet part of the group. Perhaps it would be sufficient to use something like this instead:
```java
public boolean hasMemberId() {
  return !memberId.isEmpty();
}
```